example/spec/s3Access/s3AccessSpec.js

```diff
@@ -90,36 +92,44 @@ describe('When accessing an S3 bucket directly', () => {
           path: '/s3credentials'
         }
       });
-      credentials = JSON.parse(payload.body);
+      creds = JSON.parse(payload.body);
+    });
+
+    it('the expected user can assume same region access', async () => {
+      const credentials = new Credentials(creds);
+      const sts = new STS({ credentials });
+      const whoami = await sts.getCallerIdentity().promise();
+
+      expect(whoami.Arn).toMatch(/arn:aws:sts::\d{12}:assumed-role\/s3-same-region-access-role\/userName.*/);
+      expect(whoami.UserId).toMatch(/.*:userName.*/);
     });
```

Comment:
the getCallerIdentity on the token returns something like this 
```
    {
      "Arn": "arn:aws:sts::596205514787:assumed-role/s3-same-region-access-role/userNameb4dc82c935",
      "Account": "596205514787",
      "UserId": "AROAIH4VDHQ3M6PQUWHWM:userNameb4dc82c935",
      "ResponseMetadata": {
        "RequestId": "efa6cd7c-4023-11e9-8a35-b79d7bdd24ad"
      }
    }
```
I modified the test setup to use randomId instead of randomString, so I could match the userName that was created during the call to the api.  I didn't see a way to actually retrieve the userName directly to compare the whole thing.  But this seems good enough to me.